include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -1989,30 +1992,11 @@ namespace internal
           static_dofs_per_face :
           Utilities::pow(data.data.front().fe_degree + 1, dim - 1);
 
-      // we allocate small amounts of data on the stack to signal the compiler
```

Comment:
why did you decide that this optimization is no longer necessary?

Reply:
The problem is that the allocation depends on the number of components, which is now a compile time constant. The alternative would have been to put only the data for a single component on the stack and loop in this code here rather than inside the functions (with the `/* n_components */ 1` argument). The reason why I went this way is that it reduces the amount of code slightly, plus that this part is more of the generic part where the optimization will pay off less than in the more optimized `gather_evaluate` which still has it.